src/aspire/abinitio/commonline_c3_c4.py

```diff
@@ -0,0 +1,363 @@
+import logging
+
+import numpy as np
+from numpy.linalg import eig, norm
+
+from aspire.abinitio import CLOrient3D
+
+logger = logging.getLogger(__name__)
+
+
+class CLSymmetryC3C4(CLOrient3D):
+    """
+    Define a class to estimate 3D orientations using common lines methods for molecules with
+    C3 and C4 cyclic symmetry.
+
+    The related publications are listed below:
+    G. Pragier and Y. Shkolnisky,
+    A Common Lines Approach for Abinitio Modeling of Cyclically Symmetric Molecules,
+    Inverse Problems, 35, 124005, (2019).
+
+    Y. Shkolnisky, and A. Singer,
+    Viewing Direction Estimation in Cryo-EM Using Synchronization,
+    SIAM J. Imaging Sciences, 5, 1088-1110 (2012).
+
+    A. Singer, R. R. Coifman, F. J. Sigworth, D. W. Chester, Y. Shkolnisky,
+    Detecting Consistent Common Lines in Cryo-EM by Voting,
+    Journal of Structural Biology, 169, 312-322 (2010).
+    """
+
+    def __init__(self, src, n_symm=None, n_rad=None, n_theta=None):
+        """
+        Initialize object for estimating 3D orientations for molecules with C3 and C4 symmetry.
+
+        :param src: The source object of 2D denoised or class-averaged images with metadata
+        :param n_symm: The symmetry order of the molecule. 3 or 4.
+        :param n_rad: The number of points in the radial direction
+        :param n_theta: The number of points in the theta direction
+        """
+
+        super().__init__(src, n_rad=n_rad, n_theta=n_theta)
+
+        self.n_symm = n_symm
+        self.n_ims = self.n_img
```

Comment:
Can we make it so that there isn't an `n_ims` and an `n_img`

Reply:
Changed everything to `n_img`.